th2_data_services/interfaces/adapter.py

```diff
@@ -16,7 +16,23 @@
 from typing import Any, Iterable
 
 
-class IAdapter(ABC):
+class IStreamAdapter(ABC):
+    """Interface of Adapter for streams."""
+
+    @abstractmethod
+    def handle_stream(self, stream: Iterable):
```

Comment:
I think we can rename it to just `handle` If we have separate stream adapter interface.

Reply:
I dont think it will be a good idea for 2 abstract methods to have same method name. IAdapter inherits both and it will be confusing.

Reply:
I didn't notice IAdapter. 

Do we really need IAdapter?
Honestly, It looks strange that `IAdapter` inherits `IStreamAdapter` and `IRecordAdapter` but not vice versa.

Reply:
IAdapter implements 2 abstract methods and class which will inherit IAdapter must implement those 2 methods. I tested IEventAdapter and it would let me run use it without implementing both classes. 
`TypeError: Can't instantiate abstract class {class_name} with abstract method {method_name}` error was raised.

Reply:
I just don't sure that we need both methods for the same adapter. I've never met such cases where it will useful.